### ISSUE TITLE
UnicodeDecodeError: 'ascii', Trouble with a fresh Install

### ISSUE THREAD

**Issue by petmi627** [CIRCL/AIL-framework#321]
Dear Ail Team,

I have some problems, first I want to install it with docker but I get always this error even if I run it as root,
```bash
docker build -t ail-framework .

Sending build context to Docker daemon  3.318GB
Error response from daemon: Error processing tar file(exit status 1): permission denied
```

I'm not sure but I think this errors occurs when you start it on an LXC container.

Anyway then I started an normal installation
```bash
git clone https://github.com/CIRCL/AIL-framework.git
cd AIL-framework
sudo ./installing_deps.sh
cd ~/AIL-framework/
. ./AILENV/bin/activate
#changed se_direct_io_for_flush_and_compaction to false in file configs/6382.conf
cd bin/
./LAUNCH.sh
cd var/www/
bash update_thirdparty.sh
./Flask_server.py
```
Then I start Flask_server.py, I received another Error
```python3
Misp not connected
The HIVE not connected
VT submission is disabled
Traceback (most recent call last):
  File "./Flask_server.py", line 73, in <module>
    importlib.import_module(name)
  File "/home/cyber/AIL-framework/AILENV/lib/python3.6/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 994, in _gcd_import
  File "<frozen importlib._bootstrap>", line 971, in _find_and_load
  File "<frozen importlib._bootstrap>", line 955, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 665, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "modules/Tags/Flask_Tags.py", line 36, in <module>
    clusters = Clusters(skip_duplicates=True)
  File "/home/cyber/AIL-framework/AILENV/lib/python3.6/site-packages/pymispgalaxies/api.py", line 274, in __init__
    clusters.append(json.load(f))
  File "/usr/lib/python3.6/json/__init__.py", line 296, in load
    return loads(fp.read(),
  File "/home/cyber/AIL-framework/AILENV/lib/python3.6/encodings/ascii.py", line 26, in decode
    return codecs.ascii_decode(input, self.errors)[0]
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 35947: ordinal not in range(128)
```

The script "update_thirdparty.sh" was executed without a problem.
Has anyone a Idea what it could be?

Thank you in advance for your help

**Comment by Rafiot** [comment]
Can you try to reinstall pymisp galaxies  by re-running update_thirdparty.sh? This is a weird error.

**Comment by petmi627** [comment]
Hi, No after running update_thirdparty.sh I still get the same error.

Then it help, here is a list with all Packages installed:

`adns===1.4-py1
asciimatics==1.10.0
asn1crypto==0.24.0
backcall==0.1.0
certifi==2018.11.29
cffi==1.11.5
chardet==3.0.4
Click==7.0
colorama==0.4.1
crcmod==1.7
cryptography==2.5
cycler==0.10.0
cymru-services==1.4
Cython==0.29.4
decorator==4.3.2
dnspython==1.16.0
DomainClassifier==0.7
Flask==1.0.2
future==0.17.1
idna==2.8
ipaddress==1.0.22
IPy==0.83
ipython==7.2.0
ipython-genutils==0.2.0
itsdangerous==1.1.0
jedi==0.13.2
Jinja2==2.10
jsbeautifier==1.6.2
jsonschema==2.6.0
kiwisolver==1.0.1
langid==1.1.6
Logbook==1.4.3
MarkupSafe==1.1.0
matplotlib==3.0.2
mmh3==2.5.1
networkx==2.2
nltk==3.4
numpy==1.16.1
olefile==0.43
parso==0.3.2
peepdf==0.4.2
pexpect==4.6.0
phonenumbers==8.10.4
pickleshare==0.7.5
Pillow==5.4.1
pkg-resources==0.0.0
prompt-toolkit==2.0.8
psutil==5.5.0
ptyprocess==0.6.0
pubsublogger==1.2.2
pybloomfiltermmap==0.3.15
pycountry==18.12.8
pycparser==2.19
pyfaup==1.0
pyfiglet==0.8.post0
Pygments==2.3.1
pylibinjection==0.2.4
pymisp==2.4.102
pymispgalaxies==0.1
pyparsing==2.3.1
PySocks==1.6.8
pytaxonomies==0.5
python-dateutil==2.7.5
python-Levenshtein==0.12.0
python-magic==0.4.15
pythonaes==1.0
pyzmq==17.1.2
redis==2.10.6
requests==2.21.0
SFlock==0.3.8
singledispatch==3.4.0.3
six==1.12.0
ssdeep==3.3
terminaltables==3.1.0
textblob==0.15.2
texttable==1.6.0
thehive4py==1.5.3
tlsh==0.2.0
traitlets==4.3.2
urllib3==1.24.1
wcwidth==0.1.7
Werkzeug==0.14.1
Whoosh==2.7.4
zmq==0.0.0
`

**Comment by petmi627** [comment]
I get ANSI_X3.4-1968

```python
>>> import locale
>>> locale.getpreferredencoding(False)
'ANSI_X3.4-1968'
>>> locale.getpreferredencoding()
'ANSI_X3.4-1968'

```

Well I think I try to change them to UTF-8

**Comment by Rafiot** [comment]
Woah, yeah, not sure what is going on here, but it should be 'UTF-8'

**Comment by petmi627** [comment]
HI Again,

I found something, if I get this output:

```bash
python3 -c 'import locale; print(locale.getdefaultlocale())'
(None, None)
python3 -c 'import locale; print(locale.getpreferredencoding())'
ANSI_X3.4-1968
```

I need to change the locale to en_US.UTF8 with 
```bash
export LANG="en_US.UTF-8"
```

After applying this change the output is:
```bash
python3 -c 'import locale; print(locale.getdefaultlocale())'
('en_US', 'UTF-8')
python3 -c 'import locale; print(locale.getpreferredencoding())'
UTF-8
``` 

And the problem is solved. But now I have a second Problem,  
When I start it, I get fallowing message:

```
AIL-framework virtualenv seems to exist, good
	* Launching Redis servers
	* Launching ARDB servers
	* Launching logging process
	* Launching all the queues
Could not connect to Redis at 127.0.0.1:6382: Connection refused
	6382 ARDB not ready
	Script not started, waiting 5 more secondes
```

I'm still trying to solve this Issue

**Comment by Terrtia** [comment]
Hi @petmi627,

ARDB need some time to be launched.

You just need to launch all the scripts when ardb is fully launched.
That why the launcher is waiting for ARDB. When ARDB is ready, the launcher will launch all scripts

(you can check ardb status in the `ARDB_AIL` screen.)

**Comment by petmi627** [comment]
Hi @Terrtia 

How long does it take to start ARDB normally?
Because I start the bin/LAUNCH.sh script 1 hour ago and it is still not working. I already tried to reboot it, but without success.

Thank You

**Comment by Terrtia** [comment]
Hi @petmi627 

This should take a few minutes.

Can you check the output of the ARDB_AIL screen ? Command: ``screen -r ARDB_AIL``

**Comment by petmi627** [comment]
Hi @Terrtia 

Thank you this will help, I keep you up to date

**Comment by petmi627** [comment]
Hi @Rafiot and @Terrtia,

Thank you for you help, I finally found the Issue, the Permissions where messed up, after setting the permission for the current user the problem was solved.

I got always an error that ARDB could not access the file `config/6382.conf`, I tired changing the permission only for this file but I could not solved it. So finally I changed the permission of the whole project to the current user and the problem was solved.

Again thank you very much.
